File: java_projects/glue_code/commons-validator/src/main/java/org/apache/commons/validator/routines/ShortValidator.java
package org.apache.commons.validator.routines;

import java.text.Format;
import java.util.Locale;
import org.apache.commons.validator.ContextInitializer;
import org.graalvm.polyglot.Value;

public class ShortValidator extends AbstractNumberValidator {
  private static final ShortValidator VALIDATOR = ShortValidator.ShortValidator1();
  private static final long serialVersionUID = -5227510699747787066L;
  private static Value clz =
      ContextInitializer.getPythonClass("/routines/ShortValidator.py", "ShortValidator");
  private Value obj;

  public ShortValidator(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  protected Object processParsedValue(Object value, Format formatter) {
    //
    //
    // long longValue = ((Number) value).longValue();
    //
    // if (longValue < Short.MIN_VALUE || longValue > Short.MAX_VALUE) {
    // return null;
    // }
    // return Short.valueOf((short) longValue);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("processParsedValue", value, formatter).as(Object.class);
  }

  public boolean maxValue1(Short value, short max) {
    //
    // return maxValue0(value.shortValue(), max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("maxValue1", value, max).as(boolean.class);
  }

  public boolean maxValue0(short value, short max) {
    //
    // return (value <= max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("maxValue0", value, max).as(boolean.class);
  }

  public boolean minValue1(Short value, short min) {
    //
    // return minValue0(value.shortValue(), min);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("minValue1", value, min).as(boolean.class);
  }

  public boolean minValue0(short value, short min) {
    //
    // return (value >= min);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("minValue0", value, min).as(boolean.class);
  }

  public boolean isInRange1(Short value, short min, short max) {
    //
    // return isInRange0(value.shortValue(), min, max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("isInRange1", value, min, max).as(boolean.class);
  }

  public boolean isInRange0(short value, short min, short max) {
    //
    // return (value >= min && value <= max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("isInRange0", value, min, max).as(boolean.class);
  }

  public Short validate3(String value, String pattern, Locale locale) {
    //
    // return (Short) parse(value, pattern, locale);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate3", value, pattern, locale).as(Short.class);
  }

  public Short validate2(String value, Locale locale) {
    //
    // return (Short) parse(value, (String) null, locale);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate2", value, locale).as(Short.class);
  }

  public Short validate1(String value, String pattern) {
    //
    // return (Short) parse(value, pattern, (Locale) null);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate1", value, pattern).as(Short.class);
  }

  public Short validate0(String value) {
    //
    // return (Short) parse(value, (String) null, (Locale) null);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate0", value).as(Short.class);
  }

  public static ShortValidator ShortValidator1() {
    //
    // return new ShortValidator(true, STANDARD_FORMAT);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("ShortValidator1").as(ShortValidator.class);
  }

  public ShortValidator(boolean strict, int formatType) {
    //
    // super(strict, formatType, false);
    //

    this.obj = clz.invokeMember("__init__", strict, formatType);
  }

  public static ShortValidator getInstance() {
    //
    // return VALIDATOR;
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("getInstance").as(ShortValidator.class);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/signature/commons-validator/src/main/org/apache/commons/validator/routines/ShortValidator.py
from __future__ import annotations
import time
import locale
import re
import io
import numbers
import typing
from typing import *
import datetime
from src.main.org.apache.commons.validator.routines.AbstractNumberValidator import *


class ShortValidator(AbstractNumberValidator):

    __VALIDATOR: ShortValidator = None
    __serialVersionUID: int = -5227510699747787066

    @staticmethod
    def initialize_fields() -> None:
        ShortValidator.__VALIDATOR: ShortValidator = ShortValidator.ShortValidator1()

    def _processParsedValue(
        self, value: typing.Any, formatter: typing.Union[str, datetime.datetime]
    ) -> typing.Any:

        long_value = value.value

        if long_value < -32768 or long_value > 32767:
            return None
        return short(long_value)

    def maxValue1(self, value: int, max: int) -> bool:
        return self.maxValue0(value, max)

    def maxValue0(self, value: int, max: int) -> bool:
        return value <= max

    def minValue1(self, value: int, min: int) -> bool:
        return self.minValue0(value, min)

    def minValue0(self, value: int, min: int) -> bool:
        return value >= min

    def isInRange1(self, value: int, min: int, max: int) -> bool:

        if value is None:
            return False
        if value < min or value > max:
            return False
        return True

    def isInRange0(self, value: int, min: int, max: int) -> bool:
        return value >= min and value <= max

    def validate3(self, value: str, pattern: str, locale: typing.Any) -> int:
        return int(self._parse(value, pattern, locale))

    def validate2(self, value: str, locale: typing.Any) -> int:
        return self._parse(value, None, locale)

    def validate1(self, value: str, pattern: str) -> int:
        return self._parse(value, pattern, None)

    def validate0(self, value: str) -> int:
        return self._parse(value, None, None)

    @staticmethod
    def ShortValidator1() -> ShortValidator:
        return ShortValidator(True, ShortValidator.STANDARD_FORMAT)

    def __init__(self, strict: bool, formatType: int) -> None:
        super().__init__(strict, formatType, False)

    @staticmethod
    def getInstance() -> ShortValidator:
        return ShortValidator.__VALIDATOR


ShortValidator.initialize_fields()
